cmd/pmm-managed-starlark/main_test.go

```diff
@@ -0,0 +1,157 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/percona/pmm-managed/services/checks"
+)
+
+const (
+	invalidStarlarkScriptStderr = "Error running starlark script: thread invalid starlark script: failed to execute function check: function check accepts no arguments (1 given)"
+	invalidVersionStderr        = "Error running starlark script: unsupported check version: 5"
+	memoryConsumingScriptStderr = "fatal error: runtime: out of memory"
+)
+
+var validQueryActionResult = []map[string]interface{}{
+	{"Value": "5.7.30-33-log", "Variable_name": "version"},
+	{"Value": "Percona Server (GPL), Release 33, Revision 6517692", "Variable_name": "version_comment"},
+	{"Value": "x86_64", "Variable_name": "version_compile_machine"},
+	{"Value": "Linux", "Variable_name": "version_compile_os"},
+	{"Value": "-log", "Variable_name": "version_suffix"},
+}
+
+func TestRunChecks(t *testing.T) {
+	testCases := []struct {
+		version      uint32
+		errorMessage string
+		stderr       string
+		stdout       string
+		name         string
+		script       string
+		scriptInput  []map[string]interface{}
+	}{
+
+		{
+			version:      1,
+			errorMessage: "exit status 1",
+			stderr:       invalidStarlarkScriptStderr,
+			stdout:       "",
+			name:         "invalid starlark script",
+			script:       "def check(): return []",
+			scriptInput:  validQueryActionResult,
+		},
+		{
+			version:      5,
+			errorMessage: "exit status 1",
+			stderr:       invalidVersionStderr,
+			stdout:       "",
+			name:         "invalid version",
+			script:       "def check(): return []",
+			scriptInput:  validQueryActionResult,
+		},
+		{
+			version:      1,
+			errorMessage: "exit status 2",
+			stderr:       memoryConsumingScriptStderr,
+			stdout:       "",
+			name:         "memory consuming starlark script",
+			script:       "def check(rows): return [1] * (1 << 30-1)",
+			scriptInput:  validQueryActionResult,
+		},
+		{
+			version:      1,
+			errorMessage: "signal: killed",
+			stderr:       "",
```

Comment:
We should check that "signal: killed" is in stderr

Reply:
the stderr for that case is blank